crates/runtime/src/search/full_text/connector.rs

```diff
@@ -118,6 +123,40 @@ impl FullTextConnector {
 
         first
     }
+
+    #[allow(clippy::needless_pass_by_value)]
+    fn with_indexed_stream<F>(
+        &self,
+        federated_table: Arc<FederatedTable>,
+        f: F,
+    ) -> Option<ChangesStream>
+    where
+        F: Fn(&Arc<dyn DataConnector>, Arc<FederatedTable>) -> Option<ChangesStream>,
+    {
+        let table_provider = federated_table.try_table_provider_sync()?;
+
+        let Some(indexed) = table_provider
```

Comment:
Although this works because FTS is currently the higher `IndexedTableProvider`, I would rework to explicitly use `find_index_in_table_provider<FullTextDatabaseIndex>(...`